cstrml/staking/src/lib.rs

```diff
@@ -727,26 +730,23 @@ decl_module! {
             let controller = Self::bonded(&stash).ok_or(Error::<T>::NotStash)?;
             let mut ledger = Self::ledger(&controller).ok_or(Error::<T>::NotController)?;
 
-            let stash_balance = T::Currency::free_balance(&stash);
-
-            if let Some(mut extra) = stash_balance.checked_sub(&ledger.total) {
-                extra = extra.min(max_additional);
-                // [LIMIT ACTIVE CHECK] 1:
-                // Candidates should judge its stake limit, this promise candidates' bonded stake
-                // won't exceed.
-                if <Validators<T>>::contains_key(&stash) {
-                    let limit = Self::stake_limit(&stash).unwrap_or_default();
-                    if ledger.active >= limit {
-                        Err(Error::<T>::NoWorkloads)?
-                    } else {
-                        extra = extra.min(limit - ledger.active);
-                    }
+            let mut extra = T::Currency::transferrable_balance(&stash);
```

Comment:
Do we need add more `tests` to cover the *double lock* scenario?

Reply:
I would add some tests into balance module